pyat/at/lattice/lattice_object.py

```diff
@@ -540,6 +541,15 @@ def particle(self, particle: Union[str, Particle]):
         else:
             self._particle = Particle(particle)
 
+    def set_wake_turnhistory(self):
+        """Function to reset the shape of the turn history
+        in collective elements based on the number of slices,
+        turns and bunches
+        """
+        for e in self:
+            if e.is_collective:
+                e.clear_history(ring=self)
```

Comment:
This assumes that all `Collective` elements have a `clear_history()` method. This could be enforced by adding an `@abstact_method` to the `Collective` class. I propose an another PR for that, this one is large enough!

Reply:
Right! Maybe put an issue as a reminder? Or we just add it here it is a small addition

Reply:
As you like !